tests/test_config.py

```diff
@@ -9,9 +10,28 @@
     get_hierarchical_config,
 )
 from pathlib import Path
+import builtins
+
+OPEN = builtins.open
 
 
 class TestConfig(unittest.TestCase):
+    def get_custom_mock_open(
```

Comment:
Custom mock to selectively return the right `read_data` as a function of the `path` so that the two calls to `ConfigParser().read(...)` are patched in `get_hierarchical_config`